// @springernature/mathjax-config/src/MathJaxConfig.js
"use strict";

var SpringerNature = SpringerNature || {};

SpringerNature.MathJaxConfig = {
    /* load MathML extension */
    extensions: ["mml2jax.js"],

    /* configure input and output */
    jax: ["input/TeX", "input/MathML", "output/HTML-CSS"],

    /* align display equations to the center */
    displayAlign: "left",

    /* set zoom trigger */
    menuSettings: {
        zoom: "Click"
    },

    /* configuration of the tex2jax preprocessor:
     - set delimiters for inline and display equations
     - preview while equations are processed is "TeX"
     - allow \$ to represent a literal dollar sign
     */
    tex2jax: {
        inlineMath: [["\\(", "\\)"]],
        displayMath: [['$$', '$$'], ["\\[", "\\]"]],
        preview: "TeX",
        processEscapes: true
    },

    /* configuration of the mml2jax preprocessor:
     - preview while equations are processed is "MathML"
     */
    mml2jax: {
        preview: ["MathML"]
    },

    "fast-preview": {
        disabled: true
    },

    /* configuration of the HTML-CSS output processor:
     - available fonts are STIX and TeX
     - preferred font is STIX
     - use TeX as web-based font if none of the above is available on the user's computer
     - use TeX font for image fallback mode
     */
    "HTML-CSS": {
        EqnChunk: 500,
        EqnChunkFactor: 1,
        EqnChunkDelay: 1000,
        availableFonts: ["STIX", "TeX"],
        preferredFont: ["STIX"],
        webFont: ["TeX"],
        imageFont: ["TeX"],
        styles: {
            '.MathJax_Display': {
                "margin": 0
            },
            '#MathJax_Message': {
                "margin": 0
            }
        }
    },
    TeX: {

        TagSide: "right",

        Macros: {
            bgroup: '{\\unicode{x007B}}',

            egroup: '{\\unicode{x007D}}',

            enskip: '{\\enspace}',

            lefteqn: ['\\rlap{#1}', 1],

            parbox: ['\\mbox{#2}', 2],

            upvarepsilon: '{\\unicode{x03B5}}',

            ss: '{\\unicode{x1E9E}}',

            textperthousand: '{\\unicode{x2030}}',

            textquotedblright: "{\\unicode{x201D}}",

            textregistered: '{\\unicode{0x00AE}}',

            upvartheta: '{\\unicode{x03D1}}',

            Upvartheta: '{\\unicode{x03F4}}',

            Upgamma: '{\\unicode[Times]{x0393}}',

            upgamma: '{\\unicode[Times]{x03B3}}',

            Updelta: '{\\unicode[Times]{x0394}}',

            updelta: '{\\unicode[Times]{x03B4}}',

            Uptheta: '{\\unicode[Times]{x0398}}',

            uptheta: '{\\unicode[Times]{x03B8}}',

            Upkappa: '{\\unicode[Times]{x039A}}',

            upkappa: '{\\unicode[Times]{x03BA}}',

            Uplambda: '{\\unicode[Times]{x039B}}',

            uplambda: '{\\unicode[Times]{x03BB}}',

            Upsigma: '{\\unicode[Times]{x03A3}}',

            upsigma: '{\\unicode[Times]{x03C3}}',

            Upmu: '{\\unicode[Times]{x039C}}',

            upmu: '{\\unicode[Times]{x03BC}}',

            Upiota: '{\\unicode[Times]{x0399}}',

            upiota: '{\\unicode[Times]{x03B9}}',

            Upnu: '{\\unicode[Times]{x039D}}',

            upnu: '{\\unicode[Times]{x03BD}}',

            Upxi: '{\\unicode[Times]{x039E}}',

            upxi: '{\\unicode[Times]{x03BE}}',

            Upomicron: '{\\unicode[Times]{x039F}}',

            upomicron: '{\\unicode[Times]{x03BF}}',

            Uppi: '{\\unicode[Times]{x03A0}}',

            uppi: '{\\unicode[Times]{x03C0}}',

            Uprho: '{\\unicode[Times]{x03A1}}',

            uprho: '{\\unicode[Times]{x03C1}}',

            Uptau: '{\\unicode[Times]{x03A4}}',

            uptau: '{\\unicode[Times]{x03C4}}',

            Upupsilon: '{\\unicode[Times]{x03A5}}',

            upupsilon: '{\\unicode[Times]{x03C5}}',

            Upphi: '{\\unicode[Times]{x03A6}}',

            upphi: '{\\unicode[Times]{x03C6}}',

            Upchi: '{\\unicode[Times]{x03A7}}',

            upchi: '{\\unicode[Times]{x03C7}}',

            Uppsi: '{\\unicode[Times]{x03A8}}',

            uppsi: '{\\unicode[Times]{x03C8}}',

            Upomega: '{\\unicode[Times]{x03A9}}',

            upomega: '{\\unicode[Times]{x03C9}}',

            upalpha: '{\\unicode[Times]{x03B1}}',

            upbeta: '{\\unicode[Times]{x03B2}}',

            upepsilon: '{\\unicode[Times]{x03B5}}',

            upzeta: '{\\unicode[Times]{x03B6}}',

            upeta: '{\\unicode[Times]{x03B7}}',

            permille: '{\\unicode{x2030}}',

            hfill: '{\\enspace\\enspace}',

            copyright: '{\\unicode{x00A9}}',

            dag: '{\\unicode{x2020}}',

            ddag: '{\\unicode{x2021}}',

            ointop: '{\\unicode{0x222E}}',

            P: '{\\unicode{0x00B6}}',

            lhook: '{\\hookrightarrow}',

            rhook: '{\\hookleftarrow}',

            fancyscript: ['{\\scr #1}', 1],

            varvec: ['\\pmb{#1}', 1]

        }
    },
    /* configuration of the math menu:
     - allow the user to select what font to use
     */
    MathMenu: {
        showFontMenu: true
    }
};

if (typeof module !== 'undefined') {
    module.exports = {
        defaultConfig: SpringerNature.MathJaxConfig
    };
}
